app/code/local/Mage/Core/Model/Session/Abstract/Varien.php

```diff
@@ -116,7 +116,34 @@ public function start($sessionName=null)
             $cookieParams['domain'] = $cookie->getDomain();
         }
 
-        call_user_func_array('session_set_cookie_params', $cookieParams);
+        if (Mage::getStoreConfig('web/cookie/cookie_samesite')) {
```

Comment:
An alternative would be to check if the constant exists via `defined` and then use it, but I am also fine with this.